rnsdomains/rns-manager-react
# Allow user to see price before registering a domain

## Registering
Allows a user to get to Step 1 of the registration process before connecting their wallet.

- User can increase/decrease years to see the price change
- When the user clicks "Register" if window.rLogin is not set, it will prompt the user to connect their wallet. 
  - If connected, the registration process starts, first by checking the amount of RIF the user has.
  - If the user cancels the rLogin process, it closes and the user can click again.
- If the user does not have the balance, use a shared Alert component

## Renewing
The renewal process used the removed `hasBalance` prop but was not refreshing it before renewing. So a user with 2 RIF could register a domain for 1 year and then attempt to renew it, wasting gas.

- Renew uses the shared Alert component.

## MISC
- Removes the 'hasBalance' redux property. This was not needed as the components can now listen to the response promise. 
- Breaks `getCost` and `hasEnoughRif` into their own functions. 
- Removes JSON RIF ABI from `src/app/tabs/newAdmin/domainInfo/abis.json`. This was duplicated since a more complete version of the ABI is located at `src/app/tabs/register/abis.json`
- Fixes a few propTypes errors on the files that were touched.

## Adds Cypress Testing
- Initial test for searching the landing page
- Test for interacting with the rLogin window. It does not use the mock provider yet, I'll add this in at a later PR since this one is already pretty large.
- 9 of the "changed" files are for adding Cypress.

## How to test

The best way to test is to have a wallet with gas and 4 RIF. Register a domain, first trying for 3+ years, then change to 1 year. After registering, try to renew for 2+ years to see the updated error. Using the remaining 2 RIF, renew the domain successfully.
